### ISSUE TITLE
SHAP values corrupted if calculated on Flask Web server

### ISSUE THREAD

**Issue by isenilov** [shap/shap#266]
I am building a simple Plotly Dash based web app where I would like to display SHAP dependency plots for Lightgbm classifier. However, when I try to plot the dependencies they all have some "step" like corruptions
![image](https://user-images.githubusercontent.com/20391404/45882374-a4b7b300-bdb6-11e8-929e-a0137ca4e30c.png)

The very same code running locally produces correct values and plots:
![image](https://user-images.githubusercontent.com/20391404/45882521-142da280-bdb7-11e8-9e7c-9fc603b413d2.png)

Would really appreciate if you could point on possible reason of the behaviour.

Thanks in advance.

**Comment by slundberg** [comment]
My guess is that those are correct SHAP values and that the model itself has that 
"step" behavior in it. Perhaps you have too high a learning rate (step size) when training the LightGBM model?

**Comment by isenilov** [comment]
Thanks for your answer!
Looks like your guess is right and the behaviour has nothing to do with the SHAP package.
Tried `catboost` instead of the `lightgbm` and the shap plots don't have any steps in them even if the classifier was trained within `dash` handler function.

**Comment by jpzhangvincent** [comment]
@isenilov I'm also looking into how to integrate shap values plots in the Dash app. I'm wondering what your approach is. Did you build a custom dash component for dash? Thanks!

**Comment by isenilov** [comment]
@jpzhangvincent Hi. Do not think that it is relevant to the issue, hope @slundberg would excuse me for reply here :). For the dependency graphs I used `dcc.Graph` component, just pass corresponding SHAP values and variable values to axis.
